File: src/main/java/com/hypherionmc/sdlink/core/accounts/DiscordUser.java
/*
 * This file is part of sdlink-core, licensed under the MIT License (MIT).
 * Copyright HypherionSA and Contributors
 */
package com.hypherionmc.sdlink.core.accounts;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@AllArgsConstructor(staticName = "of")
public class DiscordUser {

    private String effectiveName;
    private String avatarUrl;
    private long userId;
    private String asMention;

}

File: Common/src/main/kotlin/com/hypherionmc/sdlinkrw/api/accounts/DiscordUser.kt
package com.hypherionmc.sdlinkrw.api.accounts

/**
 * @author HypherionSA
 *
 * Represents a Discord user for universal use.
 * This allows both {@link Member} and {@link User} to be used interchangeably.
 */
data class DiscordUser(
    var effectiveName: String,
    var avatarUrl: String,
    var userId: Long,
    var asMention: String,
    var roleColor: Int
) {

    companion object {
        @JvmStatic
        fun of(name: String, avatarUrl: String, userId: Long, asMention: String, roleColor: Int): DiscordUser {
            return DiscordUser(name, avatarUrl, userId, asMention, roleColor)
        }
    }

}
